### PR TITLE
switch diff logic in proposal history.

### PR DESCRIPTION
**Pull request by pkarman**
papertrail gem stores "before" version, so switch diff logic
to always compare the version record against what comes next,
not what came before. This allows the current live record
to be included in the history.

trello: https://trello.com/c/TC59Hwop/325-look-into-comment-history-vs-admin-history-disrepancy

before:
history-fix-before

after:
history-fix-after

### CODE REVIEW

File: lib/c2_version.rb [hunk 1]
```diff
@@ -13,16 +13,12 @@ def attributes
...
-    when "create"
-      HashDiff.diff({}, attributes)
-    when "update"
-      prev = previous or fail("No previous version for #{self.pretty_inspect}")
-      HashDiff.diff(prev.attributes, attributes)
-    else
-      # not sure what makes the most sense here...
-      nil
-    end
+    next_version = self.next || live_version
```
Comment (jessieay) [inline lib/c2_version.rb]: papertrail has a method called `next` ? yikes, wish they'd picked a different name

File: lib/c2_version.rb [hunk 2]
```diff
@@ -13,16 +13,12 @@ def attributes
     end
   end
 
+  def live_version
```
Comment (jessieay) [inline lib/c2_version.rb]: can we put this below the `diff` method? this is in the spirit of:

> dependencies in a given class should always point downward. That is, the first method in a class should depend on methods further down in the class definition, and those methods should depend on methods even further down, and so on.

which is courtesy of Avdi Grimm / Ruby Tapas

File: lib/c2_version.rb [hunk 3]
```diff
@@ -13,16 +13,12 @@ def attributes
...
+    item
+  end
+
   def diff
```
Comment (jessieay) [inline lib/c2_version.rb]: yay getting rid of case statements

Comment (jessieay) [inline lib/c2_version.rb]: do we have a unit test for this? if not, should we add one? seems like that would be as valuable as the feature spec. I know this logic is mostly from papertrail and we don't want to test papertrail internals, but there was a logic change here that is important so seems like that should be tested and the unit level somehow?

**Comment by jessieay** [comment]
glad we got to the bottom of this! just one minor suggestion from me re: testing

**Comment by pkarman** [comment]
@jessieay added unit test (good call -- it was tested one level up by multiple classes that used it, but not at the unit level).

I just zapped the additional `live_version` method since it didn't really add much (I had created it while figuring out how best to fetch the original record).

**Comment by jessieay** [comment]
:rainbow: :raised_hands:

**Pull request merged**